packages/sirius-web/backend/sirius-web-sample-application/src/test/java/org/eclipse/sirius/web/sample/tests/integration/view/ViewDiagramIntegrationTests.java

```diff
@@ -0,0 +1,409 @@
+                  representation {
+                    id
+                  }
+                }
+              }
+            }
+            """;
+
+        var input = new CreateRepresentationInput(UUID.randomUUID(), this.projectId.toString(), representationDescriptionId, this.rootObjectId.toString(), "Diagram");
+        var executionInput = ExecutionInput.newExecutionInput()
+                .query(query)
+                .variables(Map.of("input", this.objectMapper.convertValue(input, new TypeReference<Map<String, Object>>() { })))
+                .build();
+        var executionResult = this.graphQL.execute(executionInput);
+        assertThat(executionResult.getErrors()).isEmpty();
+
+        try {
+            var jsonResult = this.objectMapper.writeValueAsString(executionResult.toSpecification());
+            String rawRepresentationId = JsonPath.read(jsonResult, "$.data.createRepresentation.representation.id");
+            this.representationId = UUID.fromString(rawRepresentationId);
+        } catch (JsonProcessingException | IllegalArgumentException exception) {
+            fail(exception.getMessage());
+        }
+    }
+
+    @Test
+    @DisplayName("Given a domain and a view, when a document and a representation are created, then we can subscribe to the representation")
+    public void givenDomainAndViewWhenDocumentAndRepresentationAreCreatedThenWeCanSubscribeToTheRepresentation() {
+        var configuration = new DiagramConfiguration(this.representationId.toString());
+        var input = new DiagramEventInput(UUID.randomUUID(), this.projectId.toString(), this.representationId.toString());
+        var payloadFlux = this.eventProcessorSubscriptionProvider.getSubscription(this.projectId.toString(), IDiagramEventProcessor.class, configuration, input);
+
+        Predicate<IPayload> isEmptyDiagramRefreshedEventPayload = payload -> {
+            if (payload instanceof DiagramRefreshedEventPayload diagramRefreshedEventPayload) {
+                var diagram = diagramRefreshedEventPayload.getDiagram();
+                return diagram.getNodes().isEmpty() && diagram.getEdges().isEmpty();
+            }
+            return false;
+        };
+
+        Predicate<IPayload> isInitializedPapayaDiagram = payload -> {
+            if (payload instanceof DiagramRefreshedEventPayload diagramRefreshedEventPayload) {
+                var diagram = diagramRefreshedEventPayload.getDiagram();
+                return diagram.getNodes().size() == 15 && diagram.getEdges().size() == 59;
+            }
+            return false;
+        };
+
+        StepVerifier.create(payloadFlux)
+                .expectNextMatches(isEmptyDiagramRefreshedEventPayload)
+                .then(this.initializeDiagram())
+                .expectNextMatches(isInitializedPapayaDiagram)
+                .thenCancel()
+                .verify(Duration.ofSeconds(5));
+    }
+
+    private Runnable initializeDiagram() {
```

Comment:
Minor style issue: this could perform the initialization directly (instead of returning a `Runnable`) and be invoked as `.then(this::initializeDiagram)` above.

Reply:
I'm not sure how this could be done. What would be the signature of this `then` method? The only method that I have found on `StepVerifier` which works is `Step<T> then(Runnable task)`.

<img width="668" alt="Capture d’écran 2023-03-29 à 10 41 57" src="https://user-images.githubusercontent.com/422587/228477758-9f02b60c-d01b-4c81-914a-50c55e2edfb3.png">

Reply:
This compiles fine for me:

```java
        StepVerifier.create(payloadFlux)
                .expectNextMatches(isEmptyDiagramRefreshedEventPayload)
                .then(this::initializeDiagram)
                .expectNextMatches(isInitializedPapayaDiagram)
                .thenCancel()
                .verify(Duration.ofSeconds(5));
    }

    private void initializeDiagram() {
        // Same body without the "return () -> {"
    }
```

Reply:
Could you show me the method used in StepVerifier.Step? I really can't see it even in [the Javadoc](https://projectreactor.io/docs/test/release/api/).

Reply:
`this::initializeDiagram` is type-compatible with `Runnable`, so it's the one in your screenshot.

```java
        Runnable task = this::initializeDiagram;
        StepVerifier.create(payloadFlux)
                .expectNextMatches(isEmptyDiagramRefreshedEventPayload)
                .then(task)
                .expectNextMatches(isInitializedPapayaDiagram)
                .thenCancel()
                .verify(Duration.ofSeconds(5));
    }

    private void initializeDiagram() {
    }
```